### PR TITLE
Move pure utility functions from core to loader-utils

### PR DESCRIPTION
**Pull request by Pessimistress**

### CODE REVIEW

File: modules/arrow/test/arrow-loader.spec.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import test from 'tape-promise/tape';
 import {validateLoader} from 'test/common/conformance';
 
 import {ArrowLoader, ArrowWorkerLoader} from '@loaders.gl/arrow';
-import {isBrowser, makeStreamIterator, resolvePath} from '@loaders.gl/core';
+import {isBrowser, makeStreamIterator, resolvePath} from '@loaders.gl/loader-utils';
```
[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/arrow/test/arrow-loader.spec.js]: It was exported by core before.

**Comment by Pessimistress** [comment]
> That said, this is a major tradeoff, especially as many smaller loaders may take the hit of the growing loader-utils module dependency.

I am not worried about this because these utilities are tree shakable.

To continue the conversation on https://github.com/uber-web/loaders.gl/pull/687: the main issue is that an app can have only one copy of `core`, but is less sensitive to duplicates of `loader-utils`. deck.gl's pre-built bundle has to expose `core` because it has to be the only true copy used. If we keep adding exports to `core` and have other modules import them, this pre-built interface does not have a choice but to non-discriminately export everything from the module, regardless of whether they are used by a loader.

More generically, in loaders.gl's pre-bundled ecosystem, if you are using the 3d-tiles module you import:

```html
<script src="https://unpkg.com/@loaders.gl/core@.3.7/dist/dist.min.js"></script>
<script src="https://unpkg.com/@loaders.gl/3d-tiles@1.3.7/dist/dist.min.js"></script>
```

If a utility is placed in loader-utils, because of tree shaking, it is not bundled with 3d-tiles unless it is really used. However, the core will have to bundle every export, because we don't know which loader it will be used with.

File: modules/loader-utils/src/index.js [hunk 2]
```diff
@@ -47,6 +47,36 @@ export {path};
...
+  forEach,
+  lineAsyncIterator,
+  textDecoderAsyncIterator,
+  numberedLineAsyncIterator
+} from './lib/iterator-utils/async-iteration';
+
+export {makeChunkIterator, concatenateChunksAsync} from './lib/iterator-utils/chunk-iteration';
+
+// deprecated
+export {
```
[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/loader-utils/src/index.js]: I don't mind removing them, just keeping the change non-breaking.

**Comment by ibgreen** [comment]
house keeping, this PR is quite out of date, closing for now.

**Pull request closed without merging**